### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by torfmaster**
This PR adds integration tests to the library. In this first attempt I add tests for
 - error handling for wrong types
 - a positive example
In the latter we only assert that the code compiles.

Moreover, I added a sample configuration for travis.

### CODE REVIEW

File: tests/compile-fail/derive_for_enum_fails.rs [hunk 1]
```diff
@@ -0,0 +1,10 @@
+extern crate rocket;
```
Comment (Woyten) [inline tests/compile-fail/derive_for_enum_fails.rs:1]: `extern crate` should not be used in Rust 2018.

File: rust-toolchain [hunk 2]
```diff
@@ -0,0 +1 @@
+nightly-2019-02-03
```
Comment (Woyten) [inline rust-toolchain:1]: Have you tried leaving out this line?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  - linux
+
+cache: rust
+
+install:
+  - rustup component add rustfmt-preview
+
+script:
+  - cargo fmt --all -- --check
+  - cargo build
```
Comment (Woyten) [inline .travis.yml]: What about `cargo build --all-targets`?

**Comment by Woyten** [comment]
Look good! What about other failure test cases?

**Review by Woyten — CHANGES_REQUESTED** [review]

File: tests/compile-fail/derive_for_enum_fails.rs (hunk 1 above)
Comment (torfmaster) [reply tests/compile-fail/derive_for_enum_fails.rs:1]: In this situation importing from the crate prelude does not work.

File: rust-toolchain (hunk 2 above)
Comment (torfmaster) [reply rust-toolchain:1]: Leaving out this line leads to compile errors due to unstable features used in rocket.

File: .travis.yml (hunk 3 above)
Comment (torfmaster) [reply .travis.yml]: Done.

**Comment by torfmaster** [comment]
I pushed some more negative cases. I failed on the "unexpected unnamed field", however.

File: tests/compile-fail/derive_for_generics_fails.rs [hunk 4]
```diff
@@ -0,0 +1,9 @@
+extern crate rocket;
+extern crate rocket_extra_codegen;
+
+use rocket_extra_codegen::FromRequest;
+
+#[derive(FromRequest)]
+pub struct MyUnion<T>{ //~ERROR Generics are not yet supported
```
Comment (Woyten) [inline tests/compile-fail/derive_for_generics_fails.rs]: `MyStruct`

File: tests/run-pass/derive_for_simple_struct.rs [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+use rocket_extra_codegen::FromRequest;
+
+#[derive(FromRequest)]
+#[error_type = "()"]
+pub struct BooksService {
+    _field: TestField,
+}
+
+struct TestField;
```
Comment (Woyten) [inline tests/run-pass/derive_for_simple_struct.rs:18]: You could just derive `FromRequest` for this struct.

File: tests/run-pass/derive_for_nested_struct.rs [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+#[derive(FromRequest)]
+pub struct DeepBooksService {
+    _field: TestService,
+}
+
+struct TestField;
+
+impl<'a, 'r> FromRequest<'a, 'r> for TestField {
```
Comment (Woyten) [inline tests/run-pass/derive_for_nested_struct.rs]: See below

File: tests/compile-fail/errornuous_error_type.rs [hunk 7]
```diff
@@ -0,0 +1,10 @@
+extern crate rocket;
```
Comment (Woyten) [inline tests/compile-fail/errornuous_error_type.rs]: `erroneous` :wink:

File: tests/compile_test.rs [hunk 8]
```diff
@@ -0,0 +1,20 @@
+extern crate compiletest_rs as compiletest;
```
Comment (Woyten) [inline tests/compile_test.rs:1]: Here the prelude import should work.

File: tests/compile-fail/errornuous_error_type.rs (hunk 7 above)
Comment (torfmaster) [reply tests/compile-fail/errornuous_error_type.rs]: :D

File: tests/compile_test.rs (hunk 8 above)
Comment (torfmaster) [reply tests/compile_test.rs:1]: Unfortunately, it doesn't.

**Review by Woyten — APPROVED** [review]

**Pull request merged**